src/program-spec.ts

```diff
@@ -160,11 +182,63 @@ function evalNode<Data>(root: object, node: Node<Data>): Data {
     case 'lessThan':
     case 'lessThanEq':
       return compareNodes(root, node, node.type === 'lessThanEq') as Data;
+    case 'add':
+    case 'sub':
+    case 'mul':
+    case 'div':
+      return arithmeticOperation(root, node) as Data;
     case 'and': {
       let left = evalNode(root, node.left);
       let right = evalNode(root, node.right);
       return left.and(right) as Data;
     }
+    case 'or': {
+      let left = evalNode(root, node.left);
+      let right = evalNode(root, node.right);
+      return left.or(right) as Data;
+    }
+    case 'not': {
+      let inner = evalNode(root, node.inner);
+      return inner.not() as Data;
+    }
+    case 'hash': {
+      let inner = evalNode(root, node.inner);
+      let innerFields = inner.toFields();
```

Comment:
this unfortunately doesn't work for composite types, instead we probably need to get the provable type and do `type.toFields(inner)`

Reply:
I added a TODO